### PR TITLE
Use resolvedTrainClause in codegen_alps

### LINKED ISSUE #526: Codegen_alps should get ODPS credential from db object
As we have added gomaxcompute in sqlflow, the following line should be changed to `gomaxcompute.FormatDSN(db.dataSourceName)`

https://github.com/sql-machine-learning/sqlflow/blob/95cc60823f5f5ee70aaaf5ba500775e7d592247f/sql/codegen_alps.go#L136-L141

### PR DESCRIPTION
**Pull request by uuleon**
1. Renaming `CAT_ID` to `CATAGORY_ID`
2. Use `maxcompute.config`. Fix #526 
3. Create `resolvedTrainClause` in `expression_resolver.go` and use it in `codegen_alps.go`

### CODE REVIEW

File: sql/codegen_alps.go [hunk 1]
```diff
@@ -39,137 +40,63 @@ type alpsFiller struct {
...
+	for _, css := range resolved.ColumnSpecs {
+		for _, cs := range css {
+			csCode = append(csCode, cs.ToString())
+		}
+	}
 
-	for target, columns := range pr.columns {
-		fcs, fss, err := resolveTrainColumns(&columns)
+	var odpsConfig = &gomaxcompute.Config{}
+	if db != nil {
```
Comment (tonyyang-svail) [inline sql/codegen_alps.go:83]: What happens if `db == nil`? Or, should we change to
```go
if db == nil {
        return nil, fmt.Errorf(...)
}
odpsConfig, err := gomaxcompute.ParseDSN(db.dataSourceName);
if err != nil {
        return nil, err
}
```

**Review by tonyyang-svail — APPROVED** [review]
LGTM!

File: sql/codegen_alps.go (hunk 1 above)
Comment (uuleon) [reply sql/codegen_alps.go:83]: `db == nil` happens in UnitTest

Comment (tonyyang-svail) [reply sql/codegen_alps.go:83]: @uuleon why do we need to test `db == nil`?
